### PR TITLE
[Task][68539] Init API

### PR DESCRIPTION
**Pull request by bachnd2k1**
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/68539

#### Screenshots (nếu có)

Screen Shot 2023-09-14 at 15 50 35

### CODE REVIEW

File: MovieApp/MovieApp/API/ApiCaller.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  Created by Bach Nghiem on 14/09/2023.
+//
+
+import Foundation
+
+enum APIError: Error {
+    case failedToGetData
+}
+
+class ApiCaller {
```
Comment (hlinh88) [inline MovieApp/MovieApp/API/ApiCaller.swift]:
```suggestion
final class ApiCaller {
```

File: MovieApp/MovieApp/Model/Key.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+//  Key.swift
+//  MovieApp
+//
+//  Created by Bach Nghiem on 14/09/2023.
+//
+
+import Foundation
+
+struct Key {
+    static let apiKey = "26207ea29bc9f0e10dd2556722658afb"
```
Comment (hlinh88) [inline MovieApp/MovieApp/Model/Key.swift]: Khong push api key

**Review by hlinh88 — COMMENTED** [review]
Pull co 3 commits

**Review by hlinh88 — COMMENTED** [review]
PR co conflict

**Review by hlinh88 — APPROVED** [review]

File: MovieApp/MovieApp/Model/Movie.swift [hunk 3]
```diff
@@ -7,5 +7,52 @@
...
+    let totalPages, totalResults: Int?
+
+    enum CodingKeys: String, CodingKey {
+        case dates, page, results
+        case totalPages = "total_pages"
+        case totalResults = "total_results"
+    }
+}
+
+// MARK: - Dates
```
Comment (tunghs-1253) [inline MovieApp/MovieApp/Model/Movie.swift]: Xoá đi em

**Pull request merged**